packages/neon_framework/lib/src/widgets/user_status_dialog.dart

```diff
@@ -0,0 +1,438 @@
+          builder: (final context, final statusResult) {
+            final selectedPredefinedStatus = statusResult.hasData && statusResult.requireData is user_status.Private
+                ? (statusResult.requireData as user_status.Private).messageId
+                : null;
+
+            final tiles = predefinedStatusesResult.data
+                ?.map(
+                  (final status) => Card(
+                    shape: selectedPredefinedStatus == status.id ? selectedBorder : null,
+                    child: InkWell(
+                      borderRadius: const BorderRadius.all(Radius.circular(12)),
+                      onTap: () {
+                        bloc.setPredefinedMessage(
+                          id: status.id,
+                          clearAt: status.clearAt != null
+                              ? status.clearAt.toDateTime()!.millisecondsSinceEpoch ~/ 1000
+                              : null,
+                        );
+                      },
+                      child: ListTile(
+                        leading: Text(
+                          status.icon,
+                          style: const TextStyle(fontSize: 18),
+                        ),
+                        title: Text(status.message),
+                        subtitle: Text(status.clearAt.formatRelative(context)),
+                      ),
+                    ),
+                  ),
+                )
+                .toList();
+
+            return Column(
+              children: [
+                NeonError(
+                  predefinedStatusesResult.error,
+                  onRetry: bloc.refresh,
+                ),
+                NeonLinearProgressIndicator(
+                  visible: predefinedStatusesResult.isLoading,
+                ),
+                if (tiles != null) buildGrid(NeonDialogTheme.of(context).constraints, tiles),
+              ],
+            );
+          },
+        ),
+      );
+
+  ShapeBorder get selectedBorder => RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(12),
+        side: BorderSide(
+          color: Theme.of(context).colorScheme.primary,
+          width: 1.5,
+        ),
+      );
+
+  Widget buildGrid(
+    final BoxConstraints constraints,
+    final List<Widget> children,
+  ) =>
+      Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          SizedBox(
+            width: (constraints.maxWidth - 24 * 2) / 2,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: children.sublist(0, (children.length / 2).ceil()),
+            ),
+          ),
+          SizedBox(
+            width: (constraints.maxWidth - 24 * 2) / 2,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: children.sublist((children.length / 2).ceil(), children.length),
+            ),
+          ),
+        ],
+      );
+}
```

Comment:
I'm trying to use a Wrap here to correctly display it on multiple screen sizes, but I can't get it to work at all without giving it a predefined height which of course breaks different screen sizes. @Leptopoda do you have an idea how this could work? Would it be possible to solve this mess using slivers?

Reply:
It should be enough to just make the two grids a GridView.
I don't think that slivers are really needed here (but there is also nothing against it).

Reply:
I couldn't get the GridViews to work either. The reason why I want to use Wrap is that it will wrap them on smaller screens. The grid will look weird because the tiles need the horizontal space and should avoid line breaks in the text.

Reply:
I still didn't get it to work. Can you have a look at it?